packages/expansion-hub/src/RevHub.ts

```diff
@@ -1,5 +1,18 @@
 import {RevHubType} from "./RevHubType";
+import {ExpansionHub} from "./ExpansionHub";
 
 export interface RevHub {
-    type: RevHubType
+    type: RevHubType;
+    isParent(): this is ParentRevHub;
+    isExpansionHub(): this is ExpansionHub;
+
+    readonly moduleAddress: number
+}
+
+export interface ParentRevHub extends RevHub {
+    getChildren(): ReadonlyArray<RevHub>;
```

Comment:
Getting the children does not communicate with hardware.
```suggestion
    readonly children: ReadonlyArray<RevHub>;
```

Reply:
Note that because inside of ExpansionHubInternal, children is a normal array, casting to ExpansionHubInternal does give you a mutable array, even outside of the file. We'd discussed making the children array private inside of ExpansionHubInternal.

Reply:
> One solution for this that I've implemented before is to have a private mutableChildren: RevHub[], and a public readonly children: ReadOnlyArray that is a getter over mutabeChildren, but this clutters the namespace inside of the implementation class.

I'm fine with that solution.

Reply:
Implemented the mentioned solution